### PR TITLE
fixed heading issues when shape is flipped.

### PR DESCRIPTION
**Pull request by gknisely**
@dgearhart @dnesbitt61 @kevinkreiser

### CODE REVIEW

File: src/thor/trippathbuilder.cc [hunk 1]
```diff
@@ -76,9 +76,27 @@ TripPath TripPathBuilder::Build(GraphReader& graphreader,
...
+      trip_edge->set_begin_heading(PointLL::HeadingAlongPolyline(edgeinfo->shape(),
+                                  kKmOffsetForHeading));
+      trip_edge->set_end_heading(PointLL::HeadingAtEndOfPolyline(edgeinfo->shape(),
+                                  kKmOffsetForHeading));
+
     } else {
-      trip_shape.insert(trip_shape.end(), edgeinfo->shape().rbegin() +
+
+      // Have to save the flipped shape to get the correct begin and end heading.
+      std::vector<PointLL> tmp_shape;
```
Comment (dgearhart) [inline src/thor/trippathbuilder.cc]: if you do not want to store tmp shape - you could reverse the heading for example:
begin = (end + 180) % 360

**Comment by dgearhart** [comment]
otherwise :+1:

File: src/thor/trippathbuilder.cc [hunk 2]
```diff
@@ -76,9 +76,25 @@ TripPath TripPathBuilder::Build(GraphReader& graphreader,
...
+                                  kKmOffsetForHeading));
+      trip_edge->set_end_heading(PointLL::HeadingAtEndOfPolyline(edgeinfo->shape(),
+                                  kKmOffsetForHeading));
+
     } else {
+
       trip_shape.insert(trip_shape.end(), edgeinfo->shape().rbegin() +
                     (trip_shape.size() ? 1 : 0), edgeinfo->shape().rend());
+
+      trip_edge->set_begin_heading(
```
Comment (dgearhart) [inline src/thor/trippathbuilder.cc]: does the heading get rounded off?

Comment (gknisely) [inline src/thor/trippathbuilder.cc]: was truncating

**Pull request merged**